nats.go

```diff
@@ -3944,6 +3946,13 @@ func (nc *Conn) removeSub(s *Subscription) {
 	if s.pCond != nil {
 		s.pCond.Broadcast()
 	}
+
+	// Check for watchers.
+	if jsi != nil && jsi.w != nil {
```

Comment:
This is not enough to guarantee that waitForMsgs() is not invoking the message callback at this time, or about to do it.

Reply:
Yea I am reworking this as we speak since the large KINE test failed on send to closed chan.

Reply:
ok I reworked it, please take another look.